### PR TITLE
Add SnapController actions

### PR DESCRIPTION
**Pull request by rekmarks**
Adds `SnapController` actions for methods that are used by other controllers. Adds unit tests for the same.

### CODE REVIEW

File: packages/controllers/src/snaps/SnapController.ts [hunk 1]
```diff
@@ -424,13 +487,43 @@ export class SnapController extends BaseController<
...
+  }
+
+  /**
+   * Constructor helper for registering the controller's messaging system
+   * actions.
+   */
+  private registerMessageHandlers(): void {
+    this.messagingSystem.registerActionHandler(
+      `${controllerName}:addSnap`,
+      (...args) => this.add(...args),
```
Comment (shanejonas) [inline packages/controllers/src/snaps/SnapController.ts:500]: should we name them the same as the functions, could be confusing that it the action is SnapController:addSnap but really its calling `.add`

Comment (rekmarks) [reply packages/controllers/src/snaps/SnapController.ts:500]: Yeah, that's fair. I was going back and forth but just straight up aliasing the methods probably makes the most sense.

Comment (rekmarks) [reply packages/controllers/src/snaps/SnapController.ts:500]: 1b76fc1

File: packages/controllers/src/snaps/SnapController.ts [hunk 2]
```diff
@@ -424,13 +487,43 @@ export class SnapController extends BaseController<
...
-    this._rpcHandlerMap = new Map();
+    this.registerMessageHandlers();
+  }
+
+  /**
+   * Constructor helper for registering the controller's messaging system
+   * actions.
+   */
+  private registerMessageHandlers(): void {
+    this.messagingSystem.registerActionHandler(
```
Comment (shanejonas) [inline packages/controllers/src/snaps/SnapController.ts:498]: some code golf

```typescript
['add', 'get', 'getRpcMessageHandler', 'getSnapState', 'has', 'updateSnapState'].forEach((functionName) => {
 this.messagingSystem.registerActionHandler(
      `${controllerName}:${functionName}`,
      (...args) => this[functionName](...args),
    );
})

```

**Review by shanejonas — APPROVED** [review]

File: packages/controllers/src/snaps/SnapController.ts (hunk 2 above)
Comment (rekmarks) [reply packages/controllers/src/snaps/SnapController.ts:498]: This is the way. This function should go in `@metamask/controllers`.

**Pull request merged**